src/components/combobox/combobox.tsx

```diff
@@ -1125,16 +1125,22 @@ export class Combobox
   }
 
   renderIconStart(): VNode {
-    const { selectionMode, needsIcon, selectedItems, placeholderIcon } = this;
+    const { selectedItems, placeholderIcon, selectionMode } = this;
     const selectedItem = selectedItems[0];
+
+    const hasIconAtStart = this.open
+      ? !!this.placeholderIcon
+      : selectedItem
+      ? !!selectedItem?.icon
+      : !!this.placeholderIcon;
+
     return (
-      selectionMode === "single" &&
-      needsIcon &&
-      (selectedItem?.icon || placeholderIcon) && (
+      hasIconAtStart &&
+      selectionMode === "single" && (
         <span class="icon-start">
           <calcite-icon
             class="selected-icon"
-            icon={selectedItem?.icon ?? placeholderIcon}
+            icon={!this.open && selectedItem?.icon ? selectedItem?.icon : placeholderIcon}
```

Comment:
Can you store the selected icon in a local variable? We avoid the optional chaining after the first assignment.